integrations/java/openhouse-java-runtime/src/main/java/com/linkedin/openhouse/javaclient/OpenHouseTableOperations.java

```diff
@@ -276,21 +276,21 @@ static Mono<GetTableResponseBody> handleCreateUpdateHttpError(
       return Mono.error(
           new CommitFailedException(
               casted, casted.getStatusCode().value() + " , " + casted.getResponseBodyAsString()));
-    } else if (e instanceof WebClientResponseException.GatewayTimeout
-        | e instanceof WebClientResponseException.ServiceUnavailable) {
-      return Mono.error(new CommitStateUnknownException(e));
     } else if (e instanceof WebClientResponseException.BadRequest) {
       WebClientResponseException casted = (WebClientResponseException) e;
       return Mono.error(
           new BadRequestException(
               casted, casted.getStatusCode().value() + " , " + casted.getResponseBodyAsString()));
-    } else if (e instanceof WebClientResponseException) {
+    } else if (e instanceof WebClientResponseException.Forbidden
+        || e instanceof WebClientResponseException.Unauthorized) {
       return Mono.error(new WebClientResponseWithMessageException((WebClientResponseException) e));
     } else {
       /**
-       * This serves as a catch-all for any other exceptions that are not
-       * WebClientResponseException. It helps in skipping any unexpected cleanup that could occur
-       * when a commit aborts at the caller, thus avoiding any potential data loss.
+       * This serves as a catch-all for any other WebClientResponseException exceptions including
+       * GatewayTimeouts, ServiceUgnavailable, InternalServer errors and also for any other
+       * exceptions that are not of type WebClientResponseException. This is a conservative approach
+       * to skip any unexpected cleanup that could occur when a commit aborts at the caller, thus
+       * avoiding any potential data loss.
        */
       return Mono.error(new CommitStateUnknownException(e));
```

Comment:
Is it possible to capture information about `e` here in a log and wrapper CommitStateUnknownException exception object?

Reply:
+1, it would be good to expand CommitStateUnknownException to accept and log additional params as behind the scene it is masking different types of exceptions/status code. Also this would be helpful to relate to the server side error.

Reply:
CommitStateUnknownException preserves the message returned by the throwable passed to it. 
`CommitStateUnknownException, error message: 503 Service Unavailable `
This is an example of how the error is printed at client. Do we still need to capture the status code separately?